### PR TITLE
Improve ARM64 JIT disassembly

### PR DESCRIPTION
**Pull request by TIHan**
This improves the ARM64 JIT disassembly by displaying the ext register by its correct size in 'cmp', 'add', and 'sub'.
This should handle more cases from: https://github.com/dotnet/runtime/issues/36846

Left is without the change. Right is with the change.
These are the only diffs.
before_after1
before_after2

@dotnet/jit-contrib

### CODE REVIEW

File: src/coreclr/jit/codegenarm64.cpp [hunk 1]
```diff
@@ -4557,15 +4557,12 @@ void CodeGen::genProfilingLeaveCallback(unsigned helper)
 
 // Uncomment "#define ALL_ARM64_EMITTER_UNIT_TESTS" to run all the unit tests here.
 // After adding a unit test, and verifying it works, put it under this #ifdef, so we don't see it run every time.
-//#define ALL_ARM64_EMITTER_UNIT_TESTS
+#define ALL_ARM64_EMITTER_UNIT_TESTS
```
Comment (TIHan) [inline src/coreclr/jit/codegenarm64.cpp:4560]: This will get reverted.

File: src/coreclr/jit/codegenarm64.cpp [hunk 2]
```diff
@@ -4557,15 +4557,12 @@ void CodeGen::genProfilingLeaveCallback(unsigned helper)
...
 
 #if defined(DEBUG)
 void CodeGen::genArm64EmitterUnitTests()
 {
-    if (!verbose)
```
Comment (TIHan) [inline src/coreclr/jit/codegenarm64.cpp:4565]: This will get reverted.

**Comment by TIHan** [comment]
This is ready.

**Comment by AndyAyersMS** [comment]
@BruceForstall PTAL

> This should handle more cases from: #36846

Are you planning on handling more of these cases in subsequent changes, or are we done?

**Review by BruceForstall — APPROVED** [review]

**Pull request merged**